### PR TITLE
Don't add logs coming from Sentry as breadcrumbs

### LINKED ISSUE #774: StackOverflowException on ASP.NET Core + Microsoft.Extensions.Logging with default DiagnosticLevel and Debug enabled
Please mark the type framework used:

- [ ] ASP.NET MVC
- [ ] ASP.NET Web API (OWIN)
- [x] ASP.NET Core
- [ ] WPF
- [ ] WinForms
- [ ] Xamarin
- [ ] Other:

Please mark the type of the runtime used:

- [ ] .NET Framework
- [ ] Mono
- [x] .NET Core

* Version:

Please mark the NuGet packages used:

- [x] Sentry
- [ ] Sentry.Serilog
- [ ] Sentry.NLog
- [ ] Sentry.Log4Net
- [ ] Sentry.Extensions.Logging
- [x] Sentry.AspNetCore

* Version: 3.0.0

## Description

I'm getting a `StackOverflowException` with the new version when attempting to run Integration Tests. I have my code configured to not log to Sentry if we are in unit tests by setting DSN to `""`.

If I actually Debug, it looks like I'm getting the non-stop message "Debug: Configuring the Scope".

If I downgrade to 2.1.8, this error goes away.

```
The active test run was aborted. Reason: Test host process crashed : Stack overflow.
   at Microsoft.Extensions.Logging.Debug.DebugLogger.Log[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.FormattedLogValues, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>)
   at Microsoft.Extensions.Logging.Logger.<Log>g__LoggerLog|12_0[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.ILogger, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>, System.Collections.Generic.List`1<System.Exception> ByRef, Microsoft.Extensions.Logging.FormattedLogValues ByRef)
   at Microsoft.Extensions.Logging.Logger.Log[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.FormattedLogValues, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>)
   at Microsoft.Extensions.Logging.Logger`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Microsoft.Extensions.Logging.ILogger.Log[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.FormattedLogValues, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>)
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(Microsoft.Extensions.Logging.ILogger, Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, System.Exception, System.String, System.Object[])
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(Microsoft.Extensions.Logging.ILogger, Microsoft.Extensions.Logging.LogLevel, System.Exception, System.String, System.Object[])
   at Sentry.Extensions.Logging.MelDiagnosticLogger.Log(Sentry.SentryLevel, System.String, System.Exception, System.Object[])
   at Sentry.Extensibility.DiagnosticLoggerExtensions.LogIfEnabled(Sentry.Extensibility.IDiagnosticLogger, Sentry.SentryLevel, System.String, System.Exception)
   at Sentry.Extensibility.DiagnosticLoggerExtensions.LogDebug(Sentry.Extensibility.IDiagnosticLogger, System.String)
   at Sentry.Internal.SentryScopeManager.ConfigureScope(System.Action`1<Sentry.Scope>)
   at Sentry.Internal.Hub.ConfigureScope(System.Action`1<Sentry.Scope>)
   at Sentry.SentrySdk.ConfigureScope(System.Action`1<Sentry.Scope>)
   at Sentry.Extensibility.HubAdapter.ConfigureScope(System.Action`1<Sentry.Scope>)
   at Sentry.HubExtensions.AddBreadcrumb(Sentry.IHub, Sentry.Infrastructure.ISystemClock, System.String, System.String, System.String, System.Collections.Generic.IDictionary`2<System.String,System.String>, Sentry.Protocol.BreadcrumbLevel)
   at Sentry.Extensions.Logging.SentryLogger.Log[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.FormattedLogValues, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>)
   at Microsoft.Extensions.Logging.Logger.<Log>g__LoggerLog|12_0[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.ILogger, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>, System.Collections.Generic.List`1<System.Exception> ByRef, Microsoft.Extensions.Logging.FormattedLogValues ByRef)
   at Microsoft.Extensions.Logging.Logger.Log[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.FormattedLogValues, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>)
   at Microsoft.Extensions.Logging.Logger`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].Microsoft.Extensions.Logging.ILogger.Log[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.FormattedLogValues, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>)
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(Microsoft.Extensions.Logging.ILogger, Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, System.Exception, System.String, System.Object[])
   at Microsoft.Extensions.Logging.LoggerExtensions.Log(Microsoft.Extensions.Logging.ILogger, Microsoft.Extensions.Logging.LogLevel, System.Exception, System.String, System.Object[])
   at Sentry.Extensions.Logging.MelDiagnosticLogger.Log(Sentry.SentryLevel, System.String, System.Exception, System.Object[])
   at Sentry.Extensibility.DiagnosticLoggerExtensions.LogIfEnabled(Sentry.Extensibility.IDiagnosticLogger, Sentry.SentryLevel, System.String, System.Exception)
   at Sentry.Extensibility.DiagnosticLoggerExtensions.LogDebug(Sentry.Extensibility.IDiagnosticLogger, System.String)
   at Sentry.Internal.SentryScopeManager.ConfigureScope(System.Action`1<Sentry.Scope>)
   at Sentry.Internal.Hub.ConfigureScope(System.Action`1<Sentry.Scope>)
   at Sentry.SentrySdk.ConfigureScope(System.Action`1<Sentry.Scope>)
   at Sentry.Extensibility.HubAdapter.ConfigureScope(System.Action`1<Sentry.Scope>)
   at Sentry.HubExtensions.AddBreadcrumb(Sentry.IHub, Sentry.Infrastructure.ISystemClock, System.String, System.String, System.String, System.Collections.Generic.IDictionary`2<System.String,System.String>, Sentry.Protocol.BreadcrumbLevel)
   at Sentry.Extensions.Logging.SentryLogger.Log[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.FormattedLogValues, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>)
   at Microsoft.Extensions.Logging.Logger.<Log>g__LoggerLog|12_0[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Microsoft.Extensions.Logging.ILogger, System.Exception, System.Func`3<Microsoft.Extensions.Logging.FormattedLogValues,System.Exception,System.String>, System.Collections.Generic.List`1<System.Exception> ByRef, Microsoft.Extensions.Logging.FormattedLogValues ByRef)
   at Microsoft.Extensions.Logging.Logger.Log[[Microsoft.Extensions.Logging.FormattedLogValues, Microsoft.Extensions.Logging.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]](Microsoft.Extensions.Logging.LogLevel, Microsoft.Extensions.Logging.EventId, Micr
```

### PR DESCRIPTION
**Pull request by Tyrrrz**
Applying the same treatment as with `ShouldSendEvent`.

Fixes #774

### CODE REVIEW

**Comment by Tyrrrz** [comment]
Looks like `LogCritical_SentryCategory_RecordsBreadcrumbs` thinks these logs should be added as breadcrumbs.
@bruno-garcia thoughts?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## vNext
```
Comment (bruno-garcia) [inline CHANGELOG.md:3]:
```suggestion
## Unreleased
```

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
> Looks like `LogCritical_SentryCategory_RecordsBreadcrumbs` thinks these logs should be added as breadcrumbs.
> @bruno-garcia thoughts?

Oh, yeah I did have the SDK logs as breadcrumbs in ASP.NET Core. It's an opt-in, and `DiagnosticLevel` is a way to control the loglevel. nugettrends uses that for example.

**Comment by Tyrrrz** [comment]
@bruno-garcia well, this PR was attempting to fix the stackoverflow issue by not adding Sentry's logs as breadcrumbs. It's actually not opt in (at least not on Sentry's side), but the default behavior when you use MEL and set MEL's (not Sentry's) log level to Debug.

Do you have other ideas how we can eliminate the stackoverflow?

**Pull request merged**